modules/standard/Map.chpl

```diff
@@ -341,7 +365,21 @@ module Map {
                `false` otherwise.
      :rtype: bool
     */
-    proc add(k: keyType, v: valType): bool {
+    proc add(k: keyType, v: valType): bool where !isOwnedClassType(valType) {
+      _enter();
+      if myKeys.contains(k) {
+        _leave();
+        return false;
+      }
+
+      myKeys += k;
+      vals[k] = v;
+
+      _leave();
+      return true;
+    }
+
+    proc add(k: keyType, in v: valType?): bool where isOwnedClassType(valType) {
```

Comment:
Do you think we could just consolidate to a single `add` method that has the `in` intent on `v`, here? I'm confused as to why we've avoided the `in` intent on keys and values in `add` for so long - is it to avoid making a redundant copy when passing to the associative domain?